script/main.js

```diff
@@ -0,0 +1,55 @@
+const bookList = document.querySelector('#bookList');
+const bookTitle = document.querySelector('#bookTitle');
+const bookAuthor = document.querySelector('#authorName');
+const addBook = document.querySelector('#addBook');
+const bookCollection = JSON.parse(localStorage.getItem('booklist')) ?? [];
+
+function displayBook() {
+  bookList.innerHTML = '';
+  bookCollection.forEach((book) => {
+    const p1 = document.createElement('p');
+    p1.textContent = `${book.title}`;
+    const p2 = document.createElement('p');
+    p2.textContent = `${book.author}`;
+    const indexOfBook = document.createElement('p');
+    indexOfBook.id = `bookindex${book.id}`;
+    indexOfBook.textContent = book.id;
+    indexOfBook.classList.add('hidden');
+    const removeButton = document.createElement('button');
+    removeButton.textContent = 'Remove';
+    removeButton.classList.add('deleteBtn');
+    const hr = document.createElement('hr');
+    bookList.append(p1, p2, indexOfBook, removeButton, hr);
+  });
+
+  const removeBtn = document.querySelectorAll('.deleteBtn');
+  removeBtn.forEach((elem) => {
+    elem.addEventListener('click', function removeBook() {
+      const getId = this.previousSibling.textContent;
+      const indexToRemove = bookCollection.findIndex((book) => book.id === Number(getId));
+
+      if (indexToRemove !== -1) {
+        bookCollection.splice(indexToRemove, 1);
+        localStorage.setItem('booklist', JSON.stringify(bookCollection));
+        displayBook();
+      }
+    });
+  });
+}
+
+function addBookToCollection(e) {
+  e.preventDefault();
+  const bookMaxId = Number.isFinite(Math.max(...bookCollection.map((book) => book.id)))
+    ? Math.max(...bookCollection.map((book) => book.id)) + 1 : 0;
+  const book = {
+    title: bookTitle.value,
+    author: bookAuthor.value,
+    id: bookMaxId,
+  };
+  bookCollection.push(book);
```

Comment:
Well done on your implementation! It's great to see that you have implemented unique IDs for each book and are generating them appropriately. Your attention to maintaining data integrity is commendable.

- [ ] [Optional] However, I'd like to suggest a slight improvement that could enhance your code quality. Currently, you are using the `array.push()` method to add new elements to an array. While this approach works well, I recommend considering the use of the `array.concat()` method instead. The [concat()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/concat) method of [Array](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array) instances allows you to merge two or more arrays. The key advantage here is that it doesn't modify the existing arrays; instead, it returns a new array that combines the contents of the original arrays. This can help maintain data [immutability](https://developer.mozilla.org/en-US/docs/Glossary/Immutable) and prevent unexpected side effects.

By using `array.concat()` in your context, you'll be creating a new array that includes your newly generated book while keeping the original array intact. This can contribute to a more robust and maintainable codebase. Keep up the excellent work, and happy coding! 🌟📝🚀"

Reply:
@shahadat3669  thank you very much, appreciate your valuable advice 🌹